lib/cosmos/dart/lib/dart_packet_log_writer.rb

```diff
@@ -76,7 +81,8 @@ def pre_write_entry_hook(packet)
       @file.fsync
       @sync_count = 0
     end
-    @db_queue << [packet.target_name, packet.packet_name, packet.received_time, @file_size, @packet_log_id, @sync_count]
+    time = packet.received_time ? packet.received_time : Time.now.sys
+    @db_queue << [packet.target_name, packet.packet_name, time, @file_size, @packet_log_id, @sync_count]
```

Comment:
This was killing my unit tests because the SYSTEM META packet didn't have a time stamp when it was grabbed to start a new log file. Probably not an issue in the operational system but it's safer this way because you can't create a PacketLogEntry without a time and thus the thread dies.

Reply:
As we discussed, System setup_system_meta() should be setting the time in the SYSTEM META packet. I'll revert this change and add the time in setup_system_meta.